src/components/archivesSort.tsx

```diff
@@ -0,0 +1,41 @@
+import { Link } from "gatsby"
+import * as React from "react"
+import ShowYear from "../components/showYear"
+
+interface ArchivesSortProps {
+  readonly index: number
+
+  readonly currentNode: {
+    frontmatter: {
+      readonly author: string[]
+      readonly title: string
+      readonly date: string
+    }
+  }
+  readonly preNode: {
+    readonly frontmatter: {
+      readonly date: string
+    }
```

Comment:
One other minor point that I realized I did above- since `ArchivesSort` doesn't _need_ to know anything about nodes, I removed the `node` name (similar to my comment below about descriptive variable names).

Personally, if I'm using something in line I might keep the generic graphql names like `edges` and `node`:
`data.edges.filter( edge => edge.whatever.good )` - but once I am passing it into a new component I design the prop names for what only that component knows:

```ts
// what does the `node` prop mean to an article?
data.edges.map( ({node}) => <ArticleSummary node={node} key={node.id} /> )
vs
data.edges.map( ({node}) => <ArticleSummary article={node} key={node.id} /> )